internal/utils/helm/package.go

```diff
@@ -0,0 +1,473 @@
+	if err != nil {
+		return err
+	}
+
+	// Before we install CRDs, we need to ensure that CRDs do not exist in the cluster
+	// After we install CRDs, we need to ensure that the CRDs are successfully installed into the cluster
+	if !opts.SkipCRD {
+
+		if len(crdInfo) == 0 {
+			return fmt.Errorf("CRDs not found in the envoy gateway chart")
+		}
+
+		if exist, err := detectExistCRDs(crdInfo); exist == nil || *exist {
+			if err == nil {
+				err = fmt.Errorf("found installed envoy gateway CRDs and gateway api CRDs, unable to continue installation")
+			}
+			return err
+		}
+
+		if err := installCRDs(crdInfo, ht.actionConfig); err != nil {
+			return err
+		}
+
+		if exist, err := detectExistCRDs(crdInfo); exist == nil || !*exist {
+			if err != nil {
+				return fmt.Errorf("faile to install CRDs of envoy gateway")
+			}
+			return err
+		}
+
+		if opts.OnlyCRD {
+			return nil
+		}
+	}
+
+	// Merge all values flag
+	providers := getter.All(ht.envSettings)
+	egChartValues, err := ht.valuesOpts.MergeValues(providers)
+	if err != nil {
+		return err
+	}
+
+	ht.setInstallOptions(opts)
+	release, err := ht.actionInstall.Run(egChart, egChartValues)
+	if err != nil {
+		return fmt.Errorf("failed to install envoy gateway resource: %w", err)
+	}
+
+	if opts.DryRun {
+		ht.logger.Println(release.Manifest)
+		return nil
+	}
+
+	successMarker := color.New(color.FgGreen).Add(color.Bold).Sprintf("SUCCESS")
+	ht.logger.Println(fmt.Sprintf("%s: Envoy gateway installed", successMarker))
+	return nil
+}
+
+// RunUninstall By default, we only uninstall instances of the Envoy Gateway resources.
+func (ht *PackageTool) RunUninstall(opts *PackageOptions) error {
+
+	ht.setUninstallOptions(opts)
+
+	resp, err := ht.actionUninstall.Run(opts.ReleaseName)
+	if err != nil {
+		return fmt.Errorf("failed to uninstall envoy gateway: %w", err)
+	}
+
+	if opts.DryRun {
+		ht.logger.Println(resp.Release.Manifest)
+		return nil
+	}
+
+	if opts.WithCRD {
+
+		if crdInfo, err := ht.extractCRDs(resp.Release.Chart); err != nil {
+			return err
+		} else if len(crdInfo) == 0 {
+			return fmt.Errorf("CRDs not found in the envoy gateway chart")
+		} else if _, errors := ht.actionConfig.KubeClient.Delete(crdInfo); len(errors) != 0 {
+			return fmt.Errorf("failed to delete CRDs error: %s", util.MultipleErrors("", errors))
+		}
+
+	}
+
+	if err != nil {
+		return err
+	}
+
+	successMarker := color.New(color.FgGreen).Add(color.Bold).Sprintf("SUCCESS")
+	ht.logger.Println(fmt.Sprintf("%s: Envoy gateway uninstalled", successMarker))
+	return nil
+}
+
+// setCommonValue Set the common values needed for the installation chart.
+// We are not currently considering allowing users to define the following values
+func (ht *PackageTool) setCommonValue() {
+	ht.actionInstall.CreateNamespace = true
+	ht.actionInstall.GenerateName = false
+	ht.actionInstall.Description = "envoy gateway was installed using the egctl"
+	ht.actionInstall.Namespace = "envoy-gateway-system"
```

Comment:
this is a fixed value now, imo users should be able to install EG under any namespace, this can be beneficial to EG's [multi-tenancy deployment mode](https://gateway.envoyproxy.io/v1.0.0/user/operations/deployment-mode/#multi-tenancy).

Reply:
should be configurable via `--namspace` (`-n` in short)